packages/accounts/src/kernel-zerodev/validator/base.ts

```diff
@@ -0,0 +1,52 @@
+import {concatHex, Hex} from "viem";
+import {SmartAccountSigner} from "@alchemy/aa-core/src/signer/types";
+
+export enum ValidatorMode {
+    sudo = '0x00000000',
+    plugin = '0x00000001',
+
+    //To be enabled later
+    // enable = '0x00000002',
+}
+
+export interface BaseValidatorParams {
+    validatorAddress: Hex
+    mode: ValidatorMode
+    owner: SmartAccountSigner
+}
+
+export abstract class BaseValidator {
+    readonly validatorAddress: Hex
+    mode: ValidatorMode
+    owner: SmartAccountSigner
+
+    protected constructor(params: BaseValidatorParams) {
+        this.validatorAddress = params.validatorAddress
+        this.mode = params.mode
+        this.owner = params.owner
+    }
+    abstract async signMessage(message: string | Uint8Array | Hex): Promise<Hex>
+
+    getAddress(): Hex {
+        return this.validatorAddress
+    }
+
+    async getOwner (): Promise<Hex> {
+        return this.owner.getAddress();
+    }
+    async getSignature(userOpHash: Uint8Array | string | Hex): Promise<Hex> {
```

Comment:
is this meant to be used exclusively for userOp signing?

Reply:
There actually wasn't difference. Corrected it 

Pls review https://github.com/alchemyplatform/aa-sdk/pull/26/files

I have segregated kernell and this PR so that we can release core first with base changes and then use core version here.

@moldy530

Reply:
just reviewd #26